lib/smack/Prelude.cpp

```diff
@@ -280,6 +302,17 @@ std::string getBytewisePointerStorageStr() {
   return s.str();
 }
 
+void generateIntrinsics(std::stringstream& prelude,
+			  const std::vector<unsigned> integer_sizes) {
+  const unsigned MAX_INT = *max_element(std::begin(integer_sizes), std::end(integer_sizes));
+  if (SmackOptions::BitPrecise) {
+    // bswap functions
+    for (size_t size = 16; size <= MAX_INT; size += 16) {
```

Comment:
Could we not generate bvswap functions only for the relevant bit-widths that actually appear in `INTEGER_SIZES` instead of for all that are divisible by 16?
The idea would I guess be to generate bvswap only for bit-widths that are present in `INTEGER_SIZES` and are divisible by 16.

Reply:
I changed the implementation so that the version for the bit-width 16 bits lower doesn't need to exist. The idea was to make the correctness of the definitions self evident, but we have tests :).